### PR TITLE
FEATURE: Replace hamburger dropdown with Sidebar when undock

### PR DESCRIPTION
**Pull request by tgxworld**
When the experimental Sidebar is enabled, the hamburger drop down is replaced by a sidebar drop down. A user is given the ability to dock and undock the sidebar depending on their personal preference. 

Do also note that the experimental sidebar is well, considered experimental at this point so I do not intend for the features here to be perfect. What I aim to do here is to ship the changes fast so that it can be used internally by the team to provide feedback. Custom links added by plugins and dark mode toggle has not been implemented as part of this PR as I aim to tackle it in another PR.

### Desktop

Peek 2022-07-21 17-07

### Mobile

Peek 2022-07-21 17-36

### CODE REVIEW

**Comment by tgxworld** [comment]
@lis2 @OsamaSayegh Can you please review the PR for me? Thank you!
@awesomerobot Will you be able to review the design portion for this PR? Feel free to push any additional changes that you think we should make :)

**Comment by tgxworld** [comment]
Moved to https://github.com/discourse/discourse/pull/17600

File: app/assets/javascripts/discourse/app/widgets/header.js [hunk 1]
```diff
@@ -331,6 +333,24 @@ export function attachAdditionalPanel(name, toggle, transformAttrs) {
...
 }
 
+createWidget("revamped-hamburger-menu-wrapper", {
+  buildAttributes() {
+    return { "data-click-outside": true };
+  },
+
+  html() {
+    return [
+      new ComponentConnector(this, "hamburger-menu-wrapper", {}, [], {
```
Comment (OsamaSayegh) [inline app/assets/javascripts/discourse/app/widgets/header.js:343]: I think you're already aware but we now have a better way to render a Glimmer component inside widgets that doesn't require a wrapper classic component, see https://github.com/discourse/discourse/commit/98bacbd2c6b9fe57167cd32af5eb4839b4a5d1f6 and https://github.com/discourse/discourse/commit/6c5efb61c992fb9850f900b49f5df6ca57bfff9a.

**Comment by OsamaSayegh** [comment]
Haha oops will ~~move~~ copy n paste my comments to the new PR.

**Pull request closed without merging**